pkg/rigfs/winfsys.go

```diff
@@ -445,31 +444,48 @@ func (fsys *WinFsys) RemoveAll(name string) error {
 		return fsys.removeDirAll(name)
 	}
 
-	if err := fsys.conn.Exec(fmt.Sprintf("del %s", ps.DoubleQuote(filepath.FromSlash(name)))); err != nil {
+	if err := fsys.conn.Exec(fmt.Sprintf("del %s", fsys.formatPath(name))); err != nil {
 		return fmt.Errorf("%w: remove all %s: %w", ErrCommandFailed, name, err)
 	}
 	return nil
 }
 
 func (fsys *WinFsys) removeDir(name string) error {
-	if err := fsys.conn.Exec(fmt.Sprintf("rmdir /q %s", ps.DoubleQuote(filepath.FromSlash(name)))); err != nil {
+	if err := fsys.conn.Exec(fmt.Sprintf("rmdir /q %s", fsys.formatPath(name))); err != nil {
 		return fmt.Errorf("%w: rmdir %s: %w", ErrCommandFailed, name, err)
 	}
 	return nil
 }
 
 func (fsys *WinFsys) removeDirAll(name string) error {
-	if err := fsys.conn.Exec(fmt.Sprintf("rmdir /s /q %s", ps.DoubleQuote(filepath.FromSlash(name)))); err != nil {
+	if err := fsys.conn.Exec(fmt.Sprintf("rmdir /s /q %s", fsys.formatPath(name))); err != nil {
 		return fmt.Errorf("%w: rmdir %s: %w", ErrCommandFailed, name, err)
 	}
 	return nil
 }
 
 // MkDirAll creates a directory named path, along with any necessary parents. The permission bits perm are ignored on Windows.
 func (fsys *WinFsys) MkDirAll(path string, _ FileMode) error {
-	if err := fsys.conn.Exec(fmt.Sprintf("mkdir -p %s", ps.DoubleQuote(filepath.FromSlash(path)))); err != nil {
+	if err := fsys.conn.Exec(fmt.Sprintf("mkdir -p %s", fsys.formatPath(path))); err != nil {
 		return fmt.Errorf("%w: mkdir %s: %w", ErrCommandFailed, path, err)
 	}
 
 	return nil
 }
+
+// formatPath takes a path, either in windows\format or unix/format and returns a windows\format path.
+func (fsys *WinFsys) formatPath(elem ...string) string {
```

Comment:
Is that even necessary?

https://github.com/k0sproject/rig/issues/108#issuecomment-1600653418:
> On the other hand: should there even be any path normalization in that function? Wouldn't callers pass in Windows style paths when connecting to windows machines anyways?

Reply:
I guess it could be left to the user, but it may also be handy if something like `h.Fsys().Stat(path.Join(certPath, "foo.crt"))` works regardless of the remote os.

Reply:
That would actually work without any extra tricks, as Windows treats slashes as path separators, too.

Reply:
Hmmmm 🤔 . If certPath is `c:\foo\certs` and thus the result of `path.Join` is `c:\foo\certs/foo.crt`, I would be surprised if `del "c:\foo\certs/foo.crt"` actually works (does `del c:/foo/certs/foo.crt` work nowadays?)

Reply:
Well, it doesn't work with the `formatPath` function either because it will just think it's already \ separated and spits it out as is.

Reply:
Windows will treat slashes simply the same way as backslashes. So all of your examples should just work, no matter if there's slashes or backslashes or a mix of them.

Reply:
![image](https://github.com/k0sproject/rig/assets/224971/dc7baabe-11a0-4245-88d9-cda06af2e3d1)

Reply:
That's a CLI parsing error, not a file system thingy. `del` thinks that /foo is a flag. Did you try to quote it?

Reply:
This is running CLI commands on remote windows hosts. With quotes it mostly works, but for example `del "C:\foo/bar"` gives `not found: c:\foo\foo\bar` for some reason.

Reply:
Okay. The important piece that I was missing is is for once what the win32 file system APIs accept (they just work with slashes) and what the Windows userland is capable of parsing from the command line (well, mixed results).

Reply:
I'm still unsure if rig should do those path manipulations under the hood. Those are a can of worms and highly depend on the use case. Doing additional stuff "under the hood" was the reason why `filepath.Join` was [broken](https://github.com/golang/go/issues/23467) on Windows for years when used with namespaced paths (it is calling `filepath.Clean` on the joined path).

Reply:
Removed manipulations.